### DISCUSSION TITLE
uploading BOM XML file with curl only creates 10 components of 47 in the BOM file

### DISCUSSION THREAD

**Discussion by gurucubano** [DependencyTrack/dependency-track#1589] [Q&A] [answered]
I have a simple BOM XML file with 47 components. When I upload this with
```
curl \
    -X "POST" "https://dtrack-api-m1.shr.xxx.org/api/v1/bom" \
    -H 'Content-Type: multipart/form-data' \
    -H "X-Api-Key: xxxxxxxxxxxxxx" \
    -F "autoCreate=true" \
    -F "projectName=sisis-pap" \
    -F "bom=@sisis-pap.xml"
```
to a v4.4.1 server, the project `sisis-pap` gets created, but only 10 of the 47 components show up.
The BOM file is here for inspection or tests: [http://www.unixarea.de/sisis-pap.xml](http://www.unixarea.de/sisis-pap.xml)

**Comment by stevespringett** [comment, marked as answer]
The BOM is invalid.  Line 44 reads:

```xml
<component type="library" />
```

Every component must have a name.

**Comment by gurucubano** [comment]
Thanks. Fixing this in the BOM file, imports now all 47 components. The issue was caused by a formally correct CSV file which has for one item a **two** line entry:
```
...
9,Sablot-1.0.3.tar.gz,1.0.3,shipped,sisis-pap/licenses/sablotron/README,,
10,freetype-2.3.7.tar.gz,2.3.7,shipped,sisis-pap/licenses/Freetype/LICENSE.TXT

sisis-pap/licenses/Freetype/GPL.TXT,,
11,zlib-1.2.8.tar.gz,1.2.8,shipped,sisis-pap/licenses/ZLibrary/zlib.h,,
...
```
In the entry number 10 about `freetype` the field about `licenses` has multi lines, the `,` (comma) is the field separator. From this **empty** line the tool `cyclonedx-linux-xl` constructed with the following call the broken BOM XML file:
```
cyclonedx-linux-x64 convert --input-format csv --output-format xml < sisis-pap.csv > sisis-pap.xml
```
Lesion learned: one mus check the output of `cyclonedx-linux-x64`.

**Reply by msymons** [reply]
@gurucubano,  `cyclonedx-cli` can be used to perform validation although, unfortunately, the current version (0.24.0) won't tell you the line that failed validation...  but there is an enhancement issue logged to add that functionality.
